src/main/kotlin/com/atlassian/performance/tools/virtualusers/LoadTest.kt

```diff
@@ -58,18 +56,31 @@ internal class LoadTest(
     private val load = behavior.load
 
     fun run() {
+        val users = generateUsers()
         logger.info("Holding for ${load.hold}.")
         Thread.sleep(load.hold.toMillis())
         workspace.toFile().ensureDirectory()
         setUpJira()
-        applyLoad()
+        applyLoad(users)
         val nodesDump = workspace.resolve("nodes.csv")
         nodesDump.toFile().bufferedWriter().use {
             nodeCounter.dump(it)
         }
         logger.debug("Dumped node's counts to $nodesDump")
     }
 
+    private fun generateUsers(): List<User> {
+        val pool = Executors.newCachedThreadPool { runnable ->
+            Thread(runnable, "user-generation-${runnable.hashCode()}")
+        }
+        val deadline = now() + behavior.maxOverhead
```

Comment:
Naturally, `maxOverhead` is not entirely dedicated to user generation, but it's still a much better timeout than infinity or a hardcoded value. We can refine it in the future if we'll have good estimates for other overhead sources.